rotkehlchen/data/import_utils.py

```diff
@@ -0,0 +1,341 @@
+import logging
+from collections import Counter, defaultdict
+from itertools import groupby
+from typing import Any, Dict, List, Optional
+
+from rotkehlchen.accounting.ledger_actions import LedgerAction, LedgerActionType
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.assets import A_USD
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.db.ledger_actions import DBLedgerActions
+from rotkehlchen.errors.price import NoPriceForGivenTimestamp
+from rotkehlchen.exchanges.data_structures import AssetMovement, Trade
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.price import PriceHistorian
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import (
+    AssetAmount,
+    AssetMovementCategory,
+    Fee,
+    Location,
+    Price,
+    Timestamp,
+    TradeType,
+)
+
+BinanceCsvRow = Dict[str, Any]
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class BinanceEntry:
+    """This is a base BinanceEntry class
+    All row combinations in the Binance csv have to be inherited from this base class
+    """
+
+    db: DBHandler
```

Comment:
I initialize them this way to be able to pass `db` object to the root `BinanceEntry` class and reuse it in children classes. If I change the approach, it would become necessary to pass 1 more argument to each entry class but I if it's a better approach, I'll do it of course.